test/integration/base_integration_test.cc

```diff
@@ -54,6 +54,9 @@ BaseIntegrationTest::BaseIntegrationTest(const InstanceConstSharedPtrFn& upstrea
       version_(version), upstream_address_fn_(upstream_address_fn),
       config_helper_(version, *api_, config),
       default_log_level_(TestEnvironment::getOptions().logLevel()) {
+#ifdef ENVOY_USE_NEW_HTTP1_PARSER_IN_INTEGRATION_TESTS
+  config_helper_.addRuntimeOverride("envoy.reloadable_features.enable_new_http1_parser", "true");
```

Comment:
Codec impl test has two build modes now (legacy_codec_impl_test and codec_impl_test). I could add manual friends (conn pool comes to mind)

I'm not a fan of this solution in all honest and would prefer a define directly in the parser code so that I could just flip on the new parser at build time with a define, but it doesn't seem to typically be the pattern...